lib/Models/Catalog/CatalogReferences/MagdaReference.ts

```diff
@@ -213,7 +221,21 @@ export default class MagdaReference extends AccessControlMixin(
     });
   }
 
-  private static createMemberFromRecord(
+  private static overrideRecordAspects(
+    record: JsonObject | undefined,
+    override: JsonObject | undefined
+  ) {
+    if (record && override && isJsonObject(override.aspects)) {
+      if (isJsonObject(record.aspects)) {
+        for (let key in override.aspects)
+          record.aspects[key] = override.aspects[key];
+      } else {
+        record.aspects = override.aspects;
+      }
+    }
+  }
```

Comment:
I think this approach is going against our Traits/Strata system. Also the `override` trait (https://github.com/TerriaJS/terriajs//blob/4e29fbb3519058ed4be8319359b463a6ec810caa/lib/Traits/TraitsClasses/MagdaReferenceTraits.ts#L30-L36) is being used for two different things:

- replacing record.aspects in `overrideRecordAspects` - here it is being treated as Magda JSON (with `aspects` property)
- added to `override` stratum (in [`createGroupFromRecord`](https://github.com/TerriaJS/terriajs//blob/347e4e192abc5700bdc89c037e5d0f3b705a854f/lib/Models/Catalog/CatalogReferences/MagdaReference.ts#L507), [`createMemberFromTerriaAspect`](https://github.com/TerriaJS/terriajs//blob/347e4e192abc5700bdc89c037e5d0f3b705a854f/lib/Models/Catalog/CatalogReferences/MagdaReference.ts#L568) and [`createMemberFromDistributionFormat`](https://github.com/TerriaJS/terriajs//blob/347e4e192abc5700bdc89c037e5d0f3b705a854f/lib/Models/Catalog/CatalogReferences/MagdaReference.ts#L676)) - here it is being treated as Terria JSON (with traits - similar to `itemProperties`)

I think that the `override` trait should just be used for the latter (treated as TerriaJSON).

Which values in aspects do you need to override, is it just the `terria.type` property? 
If so, I'd prefer adding another trait (called `overrideType` or something)

For example (from your test)

Instead of this

```ts
// The aspects will be added to the record and used.
const theOverriddenAspects = {
  aspects: {
    terria: {
      type: theType,
      definition: {
        name: theCatalogItemName,
        url: theDataUrl
      },
      id: theMagdaItemId
    }
  }
};

// Simulate a catalog item of magda type.
const theCatalogItem = {
  id: theMagdaItemId,
  name: theCatalogItemName,
  recordId: theRecordId,
  url: "https://a.magda.portal", // ok not being used in the test
  override: theOverriddenAspects,
  type: "magda" // ok not being used in the test
};
```

You could do this

```ts
const theCatalogItem = {
  id: theMagdaItemId,
  name: theCatalogItemName,
  recordId: theRecordId,
  url: "https://a.magda.portal", // ok not being used in the test
  override: {
    name: theCatalogItemName,
    url: theDataUrl
  },
  overrideType: theType,
  type: "magda" // ok not being used in the test
};
```

Reply:
Use `addOrOverrideAspects` trait to pass on `terria` aspect.